crates/optimism/txpool/src/supervisor/errors.rs

```diff
@@ -7,17 +89,157 @@ pub enum InteropTxValidatorError {
     #[error("inbox entry validation timed out, timeout: {0} secs")]
     Timeout(u64),
 
+    /// Message does not satisfy validation requirements
+    #[error(transparent)]
+    InvalidEntry(#[from] InvalidInboxEntry),
+
     /// Catch-all variant.
     #[error("supervisor server error: {0}")]
     Other(Box<dyn error::Error + Send + Sync>),
 }
 
+/// Invalid inbox entry
+#[derive(thiserror::Error, Debug)]
+pub enum InvalidInboxEntry {
```

Comment:
I think that a lot of errors doesn't correspond to the entries itself, and rather are supervisor errors
For example:
```
UninitializedChainDatabase
NoParentForFirstBlock
```
Also i don't see the error related to the https://specs.optimism.io/interop/supervisor.html#supervisor_checkaccesslist-contents in spec, do you why is that @emhane?

Reply:
nice thanks, will open spec issue!

yes, aware some of the errors will prove redundant latest when we've completed the rust impl of the supervisor, but already by viewing the metrics for returned error codes for a while we will likely observe which errors aren't used. that, or we get a confirmation from original authors of the go impl which errors were created only with intention to use internally, whichever happens first. it was correct to include the ones we suspected could be public protocol errors from the existing list of errors in the go implementation, in order to make incremental progress. will open a spec issue to review and remove redundant error codes so it's parked ready for later.

Reply:
My point was that the name `InvalidInboxEntry` does not fit in here IMO (because most of the errors doesn't have anything to do with the entries) and maybe we need to rename the error

Reply:
let's keep the name and instead address this on specs end https://github.com/ethereum-optimism/specs/issues/686